File: REST_module/src/main/java/com/rzk/nurdor/security/PasswordUtils.java
package com.rzk.nurdor.security;

import org.mindrot.jbcrypt.BCrypt;

public class PasswordUtils {

    public static String hashPassword(String plainPassword) {
        return BCrypt.hashpw(plainPassword, BCrypt.gensalt());
    }

    public static boolean verifyPassword(String plainPassword, String hashedPassword) {
        return BCrypt.checkpw(plainPassword, hashedPassword);
    }

    public static void main(String[] args) {
        String hashedPassword = hashPassword("Password123!");
        System.out.println(hashedPassword);
    }
}

File: RMA_Project_v2/app/src/main/java/com/example/rma_nurdor_project_v2/utils/PasswordUtils.kt
package com.example.rma_nurdor_project_v2.utils
import org.mindrot.jbcrypt.BCrypt
object PasswordUtils {

    fun hashPassword(plainPassword: String): String {
        return BCrypt.hashpw(plainPassword, BCrypt.gensalt())
    }

    fun verifyPassword(plainPassword: String, hashedPassword: String): Boolean {
        return BCrypt.checkpw(plainPassword, hashedPassword)
    }
}